### ISSUE TITLE
[BUG][kotlin-spring] Doesn't create valid model when inheritance is used ($allOf)

### ISSUE THREAD

**Issue by d0x** [OpenAPITools/openapi-generator#3587]
##### Description

When using inheritance, the generated code won't compile. It fails with this error:

> 'xxx' hides member of supertype 'YYY' and needs 'override' modifier

You can see it in this example:

```kotlin
data class Pet(
        @JsonProperty("name") val name: kotlin.String? = null,
        @JsonProperty("weight") val weight: java.math.BigDecimal? = null
                                 // ^^^^^^ 'weight' hides member of supertype 'Animal' and needs 'override' modifier
) : Animal {}
```

##### openapi-generator version

Generator Version 4.0.3
Spec Version 3.0.1

##### OpenAPI declaration file content or url

```yaml
openapi: 3.0.1
info:
  title: example
  version: 1.0.0
paths:
  /test:
    get:
      responses:
        200:
          description: successful operation
components:
  schemas:
    Animal:
      type: object
      properties:
        weight:
          type: number
      discriminator:
        propertyName: animalType
      required:
        - animalType
    Pet:
      allOf:
        - $ref: "#/components/schemas/Animal"
        - type: object
          properties:
            name:
              type: string
```

##### Command line used for generation

```bash
openapi-generator generate -i myspec.yml -g kotlin-spring \
        -Dmodels \
        -Dapis \
        --additional-properties=interfaceOnly=true \
        --additional-properties=gradleBuildFile=false \
        --additional-properties=exceptionHandler=false \
        --additional-properties=sourceFolder=src/gen/kotlin
```

##### Steps to reproduce

Generate the spec above and check the Pet class.

##### Related issues/PRs

There is a similar issue which was fixed in 4.0.0.beta and openapi version 2.
But it seems like openapi version 3 breaks it.

https://github.com/OpenAPITools/openapi-generator/issues/1205

**Comment by eliezio** [comment]
I tried to reproduce the given example and it seems it's already fixed for `kotlin-spring` generator v5.4.0.
@d0x can you pls confirm?

**Comment by d0x** [comment]
Thanks! Seems to be fixed. I tried the steps to reproduce and it compiles well. 👍🏻
